www/widgets/DemoExampleMaker.js

```diff
@@ -291,6 +310,47 @@ class DemoExampleMaker extends Widget {
     this.$('[name="dem-url"]').style.display = 'block'
   }
 
+  _uploadJSON (ex) {
+    for (let step = this._data.steps.length; step >= 1; step--) {
+      this._updateStep(step, 'remove')
+    }
+    this._data = {
+      name: ex.name,
+      tags: ex.tags || null,
+      toc: ex.toc || true,
+      layout: ex.layout || 'dock-left',
+      key: Number(ex.key),
+      code: ex.code,
+      steps: ex.info ||
+      [{ title: 'getting started', focus: null, text: 'in this step...' }]
+    }
+    console.log(ex.toc)
+    this.$('[name="dem-demo-name"]').value = ex.name
```

Comment:
let's remove this `console.log()`

also, though the default "dock-left" layout is likely to be up most of the time, i think it could be helpful to check what layout netnet is in here && adjust accordingly (tiny "edge case" but as i was testing sometimes netnet was in the "welcome" layout which means i couldn't see the code after uploading it), so something like

```js
    if (NNW.layout !== this._data.layout) {
      NNW.layout = this._data.layout
    }
```